### PR TITLE
feat(framework) Add `PullServerAppInputs` and `PushServerAppOutputs` rpcs to `Driver` service

### PR DESCRIPTION
**Pull request by panh99**

### CODE REVIEW

File: src/proto/flwr/proto/driver.proto [hunk 1]
```diff
@@ -56,3 +65,18 @@ message PullTaskResRequest {
   repeated string task_ids = 2;
 }
 message PullTaskResResponse { repeated TaskRes task_res_list = 1; }
+
+// PullServerAppIns messages
+message PullServerAppInsRequest { uint64 run_id = 1; }
+message PullServerAppInsResponse {
```
Comment (jafermarq) [inline src/proto/flwr/proto/driver.proto:71]: worth keeping the same order as in https://github.com/adap/flower/blob/4970e429b35a9f1d1300427ed399330c6d08f625/src/proto/flwr/proto/clientappio.proto#L52? (excluding `Message`).

File: src/proto/flwr/proto/driver.proto [hunk 2]
```diff
@@ -40,6 +41,14 @@ service Driver {
 
   // Get FAB
   rpc GetFab(GetFabRequest) returns (GetFabResponse) {}
+
+  // Pull ServerApp inputs
+  rpc PullServerAppIns(PullServerAppInsRequest)
```
Comment (jafermarq) [inline src/proto/flwr/proto/driver.proto:46]: better this than `PullServerAppInputs`? (same applies below to `Outs`)

File: src/proto/flwr/proto/driver.proto (hunk 1 above)
Comment (chongshenng) [reply src/proto/flwr/proto/driver.proto:71]: I agree, let's keep the order for symmetry:
```python
message PullServerAppInsResponse {
  Context context = 1;
  Run run = 2;
  Fab fab = 3;
}
```

**Review by chongshenng — CHANGES_REQUESTED** [review]
Thanks @panh99. I was under the impression we should first mirror the naming as in `ClientAppIo`, so not using `*Process*` in the names. Then rename everything later in another PR?

**Review by jafermarq — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Review by chongshenng — APPROVED** [review]

**Pull request merged**